spot_rl_experiments/spot_rl/utils/geometry_utils.py

```diff
@@ -134,3 +137,117 @@ def is_pose_within_bounds(
+        z_dist_threshold (float): Threshold for z distance (in meters)
+
+    Returns:
+        is_within (bool): True if the test pose is within both linear and angular bounds of the target pose, False otherwise.
+    """
+    # Linear xy dist bounds
+    is_within_linear_xy_bounds = (
+        euclidean(test_position[:2], target_position[:2]) < xy_dist_threshold
+    )
+
+    # Linear z dist bounds
+    is_within_linear_z_bounds = (
+        abs(test_position[2] - target_position[2]) < z_dist_threshold
+    )
+
+    return bool(is_within_linear_xy_bounds and is_within_linear_z_bounds)
+
+
+def interpolation_between_two_positions(point1, point2, steps, kind="linear"):
+    """
+    Interpolate between two 3D points
+
+    Args:
+        point1 (tuple): The first 3D point
+        point2 (tuple): The second 3D point
+        steps (int): The number of steps to interpolate
+        kind (str): The type of interpolation to use (default: linear)
+
+    Returns:
+        interpolated_points (List[Tuple]): A list of interpolated points
+    """
+    # Define the two 3D points
+    x1, y1, z1 = point1
+    x2, y2, z2 = point2
+
+    # Create an array of x,y,z values for interpolation
+    x = np.array([x1, x2])
+    y = np.array([y1, y2])
+    z = np.array([z1, z2])
+
+    # Create a 1d interpolation object object for each dimension
+    f_x = interp1d(x, y, kind=kind)
+    f_y = interp1d(x, y, kind=kind)
+    f_z = interp1d(x, z, kind=kind)
+
+    # Generate the interpolated points
+    interpolated_points = []  # type: List[Tuple]
+    for t in np.linspace(x1, x2, steps):
+        interpolated_x = float(f_x(t))
+        interpolated_y = float(f_y(t))
+        interpolated_z = float(f_z(t))
+        interpolated_points.append((interpolated_x, interpolated_y, interpolated_z))
+
+    # Return the interpolated points
+    return interpolated_points
+
+
+def generate_intermediate_point(point1, point2):
+    """
+    Generate an intermediate point between two 3D points
+
+    Args:
+        point1 (tuple): The first 3D point
+        point2 (tuple): The second 3D point
+
+    Returns:
+        intermediate_point (tuple): The intermediate 3D point
+    """
+    # Define the two 3D points
+    x1, y1, z1 = point1
+    x2, y2, z2 = point2
+
+    # Elevate the z coordinate
+    z_elevation = 0.2
+    # Generate the intermediate point
+    intermediate_point = ((x1 + x2) / 2, (y1 + y2) / 2, (z1 + z2) / 2 + z_elevation)
+
+    # Return the intermediate point
+    return intermediate_point
+
+
+def get_RPY_from_vector(vect):
+    """
+    Get the roll, pitch, yaw angles from a vector
+
+    Args:
+        vect (np.array): The direction vector
+
+    Returns:
+        roll, pitch, yaw (float): The roll, pitch, yaw angles in radians
+    """
+    # Get the yaw angle
+    yaw = np.arctan2(vect[1], vect[0])
+
+    # Get the pitch angle
+    pitch = np.arctan2(vect[2], np.sqrt(vect[0] ** 2 + vect[1] ** 2))
+
+    # Get the roll angle
```

Comment:
Explain why you don't change roll in comment

Reply:
Since this method tries to compute Roll, Pitch, Yaw of a vector in space w.r.t a frame of reference.. roll **cannot** be computed from a simple vector in space.
Thus setting the roll as 0.

Reply:
**Update** - We are now setting `roll=pi/2` so robot can drop object easily even if the end-effector does not face downwards (due to place target being at a position where there isn't a feasible solution with robot facing downwards)